homeassistant/components/dio_chacon/cover.py

```diff
@@ -0,0 +1,244 @@
+"""Cover Platform for Dio Chacon REV-SHUTTER devices."""
+
+import logging
+from typing import Any
+
+from dio_chacon_wifi_api.const import DeviceTypeEnum, ShutterMoveEnum
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import (
+    DOMAIN,
+    EVENT_DIO_CHACON_DEVICE_STATE_CHANGED,
+    EVENT_DIO_CHACON_DEVICE_STATE_RELOAD,
+    MANUFACTURER,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Discover and configure covers."""
+
+    data = hass.data[DOMAIN][config_entry.entry_id]
+    dio_chacon_client = data
+
+    list_devices = await dio_chacon_client.search_all_devices(
+        device_type_to_search=[DeviceTypeEnum.SHUTTER], with_state=True
+    )
+
+    if not list_devices:
+        _LOGGER.info(
+            "DIO Chacon did not setup covers because there are no devices of this type on this account %s",
+            config_entry.title,
+        )
+        return
```

Comment:
This would be something to do in `__init__.py`, but maybe the coordinator would be nice for that too, but i need more context for that

Reply:
This code is present in case I have two dio chacon accounts and one has not covers but only switches.
Without it, I had an error at startup of homeassistant because the list_devices was empty.

I don't know if a coordinator can help for this. I have to search for examples and better understand the coordinator usage.

Reply:
But we can still move that to init and query the devices in there

Reply:
Indeed, with the implementation with the coordinator, this code has disappeared.